_pages/documentation/general_docs/kconfig_build_system.md

```diff
@@ -0,0 +1,198 @@
+build options, such as ruby memory protocol. Users need to familiar with
+Kconfig system.
+
+## Build GEM5 with Kconfig Build System
+
+```bash
+scons [OPTIONS] Kconfig_command TARGET
+```
+
+Supported kconfig commands include
+
+- defconfig
+- setconfig
+- menuconfig
+- guiconfig
+- listnewconfig
+- oldconfig
+- olddefconfig
+- savedefconfig
+
+The most common use of options are `defconfig`, `setconfig` and `menuconfig`.
+
+Use `scons --help` to list these commands, and for additional information.
+
+To build gem5 with kconfig, there are now two steps.
+
+Initial configuration
+
+```bash
+scons defconfig gem5_build build_opts/ALL
+```
+
+This will create a configuration for the gem5_build build directory based on
+`build_opts/ALL`. The configuration is stored within the build directory.
+
+To build a target in a configured build directory, run a scons command similar
+to before.
+
+```bash
+scons gem5_build/gem5.opt
+```
+
+To build gem5 kconfig with customized kconfig options, the additional step
+is required between **initial configuration** and **building the target**
+
+- Directly set kconfig options
+
+```bash
+scons setconfig gem5_build USE_KVM=y
+```
+
+- set kconfig options by `menuconfig`
+
+```bash
+scons menuconfig gem5_build
+```
+
+## Details of Kconfig Commands
+
+### defconfig
+
+Set up a config using values specified in a defconfig file, or if no value is
+given, use the default. The second argument specifies the defconfig file. All
+of available gem5 defconfig files are in `build_opts` directory.
+
+A defconfig file in the build_opts directory can be implicitly specified in
+the build path via `build/<defconfig file>/`
+
+The user can also specify the defconfig FILE outside the gem5 repository.
+
+```bash
+scons defconfig gem5_build build_opts/RISCV
+```
+
+### setconfig
+
+Set values in an existing config directory as specified on the command line.
+
+For example, to enable gem5's built in systemc kernel:
+
+```bash
+scons setconfig gem5_build USE_SYSTEMC=y
+```
+
+### menuconfig
+
+Opens the menuconfig editor which will let you view and edit config values,
+and view help text. menuconfig runs in CLI.
+
+```bash
+scons menuconfig gem5_build
+```
+
+If is successful, the CLI will look like this
+
+![](/assets/img/kconfig/menuconfig.png)
+
+To view help text, for example, USE_ARM_ISA, will look like this
+
+![](/assets/img/kconfig/menuconfig_details.png)
+
+If the `gem5_build` directory is not exists. SCons will set up a build
+directory at the path `gem5_build` with default options, and then invoke
+menuconfig so you can set up its configuration.
+
+### guiconfig
+
+Opens the guiconfig editor which will let you view and edit config values,
+and view help text. guiconfig runs as a graphical application.
+
+```bash
+scons guiconfig gem5_build
+```
+
+If is successful, it will create new windows to show up like
+
+![](/assets/img/kconfig/guiconfig.png)
+
+Open guiconfig in Ubuntu 22.04 with python3-tk
```

Comment:
Is `python3-tk` a requirement? should this go in the `optional_requirements.txt` file? Are there any other new dependencies?

Reply:
It the apt install target. I will add `python3-tk` in dockerfile.

Reply:
What dockerfile? The normal use case for gem5 is not using docker.

I think this should be in the `optional-requirements.txt` file if it is not part of a default python install.

Reply:
I thinks there is no ways to install `python3-tk` via pip. For Ubuntu users, they should install `python3-tk` via `apt`.
https://www.pythonguis.com/installation/install-tkinter-linux/